protos/idl/service.proto

```diff
@@ -98,3 +116,74 @@ message Tag {
 message Metadata {
     map<string, string> key_map = 1; // key map is a dictionary of key/val strings that represent metadata
 }
+
+// Filter expression that is composed of a combination of single filters
+message FilterExpression {
+    repeated SinglePropertyFilter filters = 1;
```

Comment:
are these always and-ed together?

Reply:
Yes for now it is, but in the future when there are use cases for "OR"s, I will add a `LogicalOperator` to the SingleProperty filter